packages/core/src/StateNode.ts

```diff
@@ -220,8 +206,28 @@ export class StateNode<
     this.history =
       this.config.history === true ? 'shallow' : this.config.history || false;
 
-    this.entry = toArray(this.config.entry).slice();
-    this.exit = toArray(this.config.exit).slice();
+    const convertAction = (
```

Comment:
we also have a very similar thing in stateUtils now

Reply:
Deduped